ui/src/app/views/concept-homepage/component.ts

```diff
@@ -0,0 +1,152 @@
+import {Component, OnInit, ViewChild} from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+
+import {ConceptAddModalComponent} from 'app/views/concept-add-modal/component';
+import {ConceptTableComponent} from 'app/views/concept-table/component';
+
+
+import {
+  Concept,
+  ConceptsService,
+  Domain,
+  DomainInfo,
+  StandardConceptFilter,
+} from 'generated';
+
+interface ConceptCacheSet {
+  domain: Domain;
+
+  items: Array<Concept>;
+}
+
+@Component({
+  styleUrls: ['../../styles/buttons.css',
+    '../../styles/cards.css',
+    '../../styles/headers.css',
+    '../../styles/inputs.css',
+    '../../styles/errors.css',
+    './component.css'],
+  templateUrl: './component.html',
+})
+export class ConceptHomepageComponent implements OnInit {
+  loadingDomains = true;
+  searchTerm: string;
+  standardConceptsOnly = false;
+  searching = false;
+  currentSearchString: string;
+  searchLoading = false;
+  selectedDomain: Domain;
+  addTextHovering = false;
+
+  @ViewChild(ConceptTableComponent)
+  conceptTable: ConceptTableComponent;
+
+  @ViewChild(ConceptAddModalComponent)
+  conceptAddModal: ConceptAddModalComponent;
+
+  conceptDomainList: Array<DomainInfo> = [];
+
+  concepts: Array<Concept> = [];
+  conceptsCache: Array<ConceptCacheSet> = [];
+
+  wsNamespace: string;
+  wsId: string;
+
+  constructor(
+    private conceptsService: ConceptsService,
+    private route: ActivatedRoute,
+  ) {
+    this.wsNamespace = this.route.snapshot.params['ns'];
+    this.wsId = this.route.snapshot.params['wsid'];
+  }
+
+  ngOnInit(): void {
+    this.loadingDomains = true;
+    this.conceptsService.getDomainInfo(this.wsNamespace, this.wsId).subscribe((response) => {
+      this.conceptDomainList = response.items;
+      this.conceptDomainList.forEach((domain) => {
+        this.conceptsCache.push({
+          domain: domain.domain,
+          items: []
+        });
+      });
+      this.loadingDomains = false;
+      this.selectedDomain = this.conceptDomainList[0].domain;
+    });
+  }
+
+  openAddModal(): void {
+    this.conceptAddModal.open();
+  }
+
+  selectDomain(domain: Domain) {
+    this.selectedDomain = domain;
+    this.setConcepts();
+  }
+
+  searchButton() {
+    this.currentSearchString = this.searchTerm;
+    this.searchConcepts();
+  }
+
+  browseDomain(domain: DomainInfo) {
+    this.currentSearchString = '';
+    this.selectedDomain = domain.domain;
+    this.searchConcepts();
+  }
+
+  searchConcepts() {
+    console.log(this.standardConceptsOnly);
+    this.searching = true;
+    this.searchLoading = true;
+    let standardConceptFilter: StandardConceptFilter;
+    if (this.standardConceptsOnly) {
```

Comment:
I think at this point in development,  i might search all concepts all the time and filter on standard concepts in the ui  . Like in setConcepts() below , filter on your checkbox there.

Reply:
I don't want to do that because we cap the search at 100 results, so I don't want it to only get non-standard concepts, and then the user is left seeing no results.